go/apps/mixnet/README.md

```diff
@@ -0,0 +1,168 @@
+Mixnet over CloudProxy
+======================
+
+A mixnet router facilitates anonymous communications among a set of peers.
+Alice wants to send Msg to Bob; she encrypts, authenticates, and sends the
+message (Msg, Bob) to Molly, a mixnet router. Molly waits until she has _n-1_
+more messages from _n-1_ other peers, then transmits the messages to their
+respective destinations simultaneously, thus anonymizing Alice among a set of n
+peers. Such a service is only useful if Molly can be trusted to not divulge the
+link (Alice, Bob). This document specifies a simple mixnet built on the
+CloudProxy platform that reduces this trust to the knowledge of a public key
+and trust in the owner of that key.
+
+Security goals and threat model
+-------------------------------
+There are three principals in our protocol: the sender, recipient, and the
+policy owner. Our goal is to design a protocol that provides anonymity for
+senders without requiring the recipient to execute the protocol. The policy
+owner controls the root policy private key which is used to attest to the
+instantiation of mixnet routers using the CloudProxy platform. CloudProxy
+exposes the code to the users, as well as the operating system in which it is
+running. To instantiate a mixnet router, a TPM coupled with the hardware
+platform generates a public key, which is attested to by the policy owner by
+signing it with the private policy key. When the machine boots, the OS is
+measured and generates a private/public key pair, which is attested to by the
+TPM; finally, the mixnet code itself is measured and generates a private/public
+key pair, which is attested to by the OS. Hence, trust in the mixnet
+routers to faithfully carry out the protocol is reduced to correct
+provisioning of the policy key.
+
+We consider a global passive adversary who observes all communications between
+senders and mixnet routers, mixnet routers and other mixnet routers, and mixnet
+routers and recipients. The adversary may also send messages on channels it
+observes. Since the service is anonymous, the adversary is allowed to control
+any number of senders or recipients. The effect is that the state of that peer
+is exposed, including any and all cryptographic keys. We assume that the policy
+key was correctly provisioned; if the code implements the protocol correctly and
+is properly isolated during its execution, we claim this precludes the
+possibility of exposing a router’s state. (This is a strong claim offered here
+without proof. CloudProxy provides assurance that the expected program is
+running; assuming the code does not contain any bugs that allow it to be
+comprimised, a formal treatment of our protocol should reduce the adversary's
+control of the routers to standard cryptographic assumptions: in particular,
+CDH on elliptic curves, as well as the authenticity and privacy of the
```

Comment:
preferably "integrity and confidentiality" over "authenticity and privacy", since the first terms are more standard in security, and "privacy" has another meaning to do with personal data.